R/calcIDataConsEneBranch.R

```diff
@@ -0,0 +1,93 @@
+#' calcIDataConsEneBranch
+#'
+#' Use data to derive OPENPROM input parameter iDataConsEneBranch
+#'
+#' @return  OPENPROM input data iDataConsEneBranch
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas, Giannis Tolios
+#'
+#' @examples
+#' \dontrun{
+#' a <- calcOutput(type = "IDataConsEneBranch", aggregate = FALSE)
+#' }
+#'
+#' @importFrom dplyr %>% select mutate left_join case_when if_else arrange
+#' @importFrom tidyr pivot_wider spread gather
+#' @importFrom quitte as.quitte interpolate_missing_periods
+#' @importFrom stringr str_split
+ 
+calcIDataConsEneBranch <- function() {
+
+  x <- readSource("ENERDATA", "electricity", convert = TRUE)
+  
+  # Get time range from GAMS code
+  fStartHorizon <- readEvalGlobal(system.file(file.path("extdata", "main.gms"), package = "mrprom"))["fStartHorizon"]
+  fStartY <- readEvalGlobal(system.file(file.path("extdata", "main.gms"), package = "mrprom"))["fStartY"]
+  x <- x[, c(fStartHorizon:fStartY), ]
+  
+  # Use ENERDATA - OPENPROM mapping to extract correct data from source
+  map <- toolGetMapping(name = "prom-enerdata-consene-mapping.csv",
+                        type = "sectoral",
+                        where = "mappingfolder")
+  
+  enernames <- as.vector( str_split(map$ENERDATA..Mtoe., "\\+", simplify = TRUE) )
```

Comment:
was there a linter and package check? I think the `$` notation is not allowed in packages